sympy/integrals/meijerint.py

```diff
@@ -1660,6 +1661,18 @@ def _meijerint_indefinite_1(f, x):
         fac_ = fac * C / (b*a**((1 + c)/b))
         rho = (c + 1)/b - 1
 
+        # handle zero denominators in the above substitution
+        special_results = []
+        for den in (a, b):
+            if not den.is_Number and not den.is_nonzero:
+                zero_sol = solve(den, dict=True)
+                for sol in zero_sol:
+                    zero_res = _meijerint_indefinite_1(f.subs(sol), x)
```

Comment:
Does this internal function return an Integral object if it can't evaluate it?

Reply:
`_meijerint_indefinite_1` and `meijerint_indefinite` both return `None` on failure. But just calling `integrate` bypasses the problem with `meijerint_indefinite(x**n, x)`!

Reply:
turned out leaving it unevaluated until later was the better option than recursive integrate (see third commit message).

Reply:
I've got code that won't break the testsuite, only adding more special cases, but it turns out that just handling the division in the substitution in meijerint_indefinite_1 isn't adequate for treating all potential zero divisions. The better approach is sharing the heurisch logic. Will post that as a separate pull plus more info why the approach here can't be complete.

Reply:
That's great. I think there is also piecewise handling code in risch and manualintegrate, and possibly others, so it would be good to investigate those as well.